### ISSUE TITLE
Error parsing terraform files when regex() function is present

### ISSUE THREAD

**Issue by gone-for-coding** [bridgecrewio/checkov#3217]
**Describe the issue**
Error parsing terraform files when regex() function is present

**Examples**
```hcl
module "abc" {

  Name = "${sum([parseint(regex("[0-9]{2}$", each.value.name), 10), 50])}"

}
```

**Version (please complete the following information):**
 - Checkov Version 2.1.53

**Additional context**
When removing regex() the file is parsed correctly.

**Comment by gruebel** [comment]
@gone-for-coding thanks for reaching out. It is actually not the presence of the `regex()` function, which causes this error, it is the regex itself. `to be more precise the combination of quotes and closing curly bracket `}$"`. Will try to come up with a fix next week.

**Comment by gruebel** [comment]
was released in version `2.1.174`
